lib/common/crmcommon_private.h

```diff
@@ -135,6 +135,11 @@ bool pcmk__marked_as_deleted(xmlAttrPtr a, void *user_data);
 G_GNUC_INTERNAL
 void pcmk__dump_xml_attr(const xmlAttr *attr, GString *buffer);
 
+#if defined(PCMK__UNIT_TESTING)
+G_GNUC_INTERNAL
+void pcmk__xa_set_expand(xmlNode *target, const char *name, const char *value);
```

Comment:
> I'd rather just make things library-private than special-case unit testing

That's fine. I don't *like* it, because it sort of violates the spirit of things... I prefer keeping scopes as tight as possible. And it means that in an ideal world where everything is unit-tested, we'd have few or no static functions.

However, in the absence of a way to unit-test a static function, making them library-private is probably the lesser evil. The `defined` guards are clunky.

(Though if we really wanted, it might be possible to make a macro with a short name that expands to these CPP directives. I think clumens and I tossed around something like `PCMK__STATIC` a while back. Not sure about that kind of CPP trickery off the top of my head... it would need to expand and then get processed again.)

> `pcmk__xe_set_score()`

that's probably good

Reply:
Yeah, I considered a macro as well, but as a general principle it's a good idea to avoid any differences between unit testing and actual calls. Not always possible, but it's best to minimize it.